### PR TITLE
Bug 1173514 - Move gravatar URL function into cacheback job to reduce hashing time impact.

### PR DESCRIPTION
**Pull request by jezdez**

### CODE REVIEW

File: kuma/users/jobs.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import hashlib
+import urllib
+
+from django.conf import settings
+
+from kuma.core.jobs import KumaJob
+
+
+DEFAULT_AVATAR = getattr(settings, 'DEFAULT_AVATAR',
+                         settings.MEDIA_URL + 'img/avatar-default.png')
```
Comment (robhudson) [inline kuma/users/jobs.py]: Why does the `settings.DEFAULT_AVATAR` have a default here? lol

Comment (jezdez) [inline kuma/users/jobs.py]: I have no idea, it's.. legacy?

File: kuma/users/models.py [hunk 2]
```diff
@@ -215,6 +216,12 @@ def create_user_profile(sender, instance, created, **kwargs):
         p, created = UserProfile.objects.get_or_create(user=instance)
 
 
+@receiver(models.signals.post_save, sender=settings.AUTH_USER_MODEL)
+def invalidate_gravatar_url(sender, instance, created, **kwargs):
+    if instance.email:
+        UserGravatarURLJob().invalidate(instance.email)
```
Comment (robhudson) [inline kuma/users/models.py]: Is it possible for the email to go from having a real value to `None` and we should invalidate that also?

File: kuma/users/jobs.py (hunk 1 above)
Comment (robhudson) [inline kuma/users/jobs.py]: Do both paths exist? I would try to just have one default. That's too confusing.

**Comment by robhudson** [comment]
r+wc (and passing tests)

File: kuma/users/models.py (hunk 2 above)
Comment (jezdez) [inline kuma/users/models.py]: Ah, hm, good question, I think that should delete the entry in that case.

**Comment by jezdez** [comment]
@darkwing provided a new default avatar image

**Comment by robhudson** [comment]
:heart_eyes:

**Pull request merged**